linkerd/idle-cache/src/lib.rs

```diff
@@ -134,6 +134,13 @@ where
         Some(cached)
     }
 
+    pub fn insert(&self, key: K, value: V) -> Cached<V>
+    where
+        V: Clone,
+    {
+        self.insert_with(key, move |_| value)
+    }
+
```

Comment:
Can you explain this part of the change a little bit more? I don't understand why the existing API doesn't work for our needs:

* `insert_with` is the same as `get_or_insert_with`, right? I guess it avoids the read lock? But it's it's effectively the same: if the value exists it returns the existing value. It gets.
* Do we need a `get_or_insert()` that calls `get_or_insert_with` internally?
* I'd generally expect `insert` to return an option holding the prior value in the occupied case...